numpy/_core/src/multiarray/dtypemeta.c

```diff
@@ -156,7 +156,7 @@ PyArray_ArrFuncs default_funcs = {
 
 NPY_NO_EXPORT int
 dtypemeta_initialize_struct_from_spec(
-        PyArray_DTypeMeta *DType, PyArrayDTypeMeta_Spec *spec)
+        PyArray_DTypeMeta *DType, PyArrayDTypeMeta_Spec *spec, int priv)
```

Comment:
It's private C++ private is reserved and it was interferring, IIRC

Reply:
Ah, I see. Still think it would be good to add a comment about what arguments are and what they are used for any time it is not obvious (which it is not here!)